### PR TITLE
#159987650 Users should be able to report an article

### PR DESCRIPTION
**Pull request by emmaadesile**
#### What does this PR do?
- Implement a feature that enables users to be able to report an article

#### Description of Task to be completed?
- users can click on the report article reaction
 - users can select the reason for the report and submit

#### How should this be manually tested?
- clone the branch
- run `npm install` to install app dependencies
- login into the app
- view an article
- click on the article reaction button beside the article like/dislike icons
- click the report article reaction
- choose a reason for the report and click submit

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[159987650](https://www.pivotaltracker.com/story/show/159987650)

#### Screenshots (if appropriate)
screen shot 2018-10-26 at 5 44 50 pm

#### Questions:
N/A

### CODE REVIEW

File: src/actions/reportArticle.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          "Content-Type": "application/json"
+        }
+      }
+    )
+    .then(response => {
+      if (response.data.status === "success") {
+        dispatch(reportArticleLoading(false));
+        return dispatch(reportArticleSucess(response.data));
+      }
+      dispatch(reportArticleLoading(false));
```
Comment (AdmiralOnu) [inline src/actions/reportArticle.js]: line 41 and line 44 can be replaced with a single dispatch before the if statement on line 40.

File: src/components/ArticleReaction/ArticleReaction.module.css [hunk 2]
```diff
@@ -38,13 +38,14 @@
...
     background: transparent;
     color: #FFF;
-  }
\ No newline at end of file
+  }
+
+
+.popup button:hover {
+  background-color: #eee; 
+}
```
Comment (AdmiralOnu) [inline src/components/ArticleReaction/ArticleReaction.module.css:50]: You may want to remove commented code on line 30, and add a new line at the end of this file

File: src/components/ReportArticle/ReportArticle.module.scss [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+.reportForm {
+  display: grid;
+}
+
+@media screen and (max-width: 478px ){
+  .reportBody { 
+    max-width: 300px;
+  }
+}
```
Comment (AdmiralOnu) [inline src/components/ReportArticle/ReportArticle.module.scss]: you may wan't to add a new line here

**Review by gonuoha — APPROVED** [review]

File: src/assets/icons/close.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z"/></svg>
```
Comment (lauragift21) [inline src/assets/icons/close.svg:1]: You can add an extra line here.

File: src/components/ArticleContent/ArticleContent.js [hunk 5]
```diff
@@ -107,111 +111,143 @@ class ArticleContent extends Component {
...
-  }
+      <div
+        className={styles.commentPrompt}
+        style={promptStyle}
+        onClick={this.displayCommentBox}
+        tabIndex={0}
+        role="button"
+      >
+        <p className={styles.commentPromptIcon}>
+          <FaComment />{" "}
```
Comment (lauragift21) [inline src/components/ArticleContent/ArticleContent.js]: On line 130. You can take the empty object.

File: src/components/ArticleContent/ArticleContent.module.scss [hunk 6]
```diff
@@ -97,4 +97,4 @@ img {
 .shareCursor:hover {
   cursor: pointer;
   opacity: 1;
-}
+}
```
Comment (lauragift21) [inline src/components/ArticleContent/ArticleContent.module.scss]: At line 100, You can add a new line there too.

**Review by lauragift21 — CHANGES_REQUESTED** [review]

**Review by lauragift21 — APPROVED** [review]

**Review by johnkingzy — APPROVED** [review]
LGTM

**Pull request merged**